src/Frontend/Modules/Profiles/Actions/Activate.php

```diff
@@ -47,10 +48,10 @@ public function execute(): void
                 $this->template->assign('activationSuccess', true);
             } else {
                 // failure
-                $this->redirect(FrontendNavigation::getUrl(404));
+                throw new NotFoundHttpException();
```

Comment:
Seems like this code could be simplified with some early returns/throws

Reply:
I had been thinking about cleaning it up but in the end I didn't do it since a lot of the actions need cleaning up, but I'll do the those that have changes in this pr